src/Controller/ControllerFactory.php

```diff
@@ -156,58 +156,83 @@ public function handle(ServerRequestInterface $request): ResponseInterface
      */
     protected function getActionArgs(Closure $action, array $passedParams): array
     {
-        $args = [];
-        $reflection = new ReflectionFunction($action);
+        $resolved = [];
+        $function = new ReflectionFunction($action);
+        foreach ($function->getParameters() as $parameter) {
+            $type = $parameter->getType();
+            $typeName = $type instanceof ReflectionNamedType ? ltrim($type->getName(), '?') : null;
 
-        foreach ($reflection->getParameters() as $parameter) {
-            $position = $parameter->getPosition();
-            $hasDefault = $parameter->isDefaultValueAvailable();
+            // Check for named parameter
+            $name = $parameter->getName();
+            if (array_key_exists($name, $passedParams)) {
```

Comment:
I don't think this will ever happen as $actionArgs is always a list of and not a map/dictionary style array.

Reply:
We can leave this out. It was from the general invoker.